backend/src/main/java/backend/mulkkam/intake/service/IntakeAmountService.java

```diff
@@ -29,6 +35,11 @@ public void modifyTarget(
     ) {
         Member member = getMember(memberId);
         member.updateTargetAmount(intakeTargetAmountModifyRequest.toAmount());
+
+        Optional<IntakeHistory> intakeHistory = intakeHistoryRepository.findByMemberIdAndHistoryDate(memberId,
+                LocalDate.now());
+        intakeHistory.ifPresent(
+                history -> history.modifyTargetAmount(new Amount(intakeTargetAmountModifyRequest.toAmount().value())));
```

Comment:
history -> history.modifyTargetAmount(intakeTargetAmountModifyRequest.amount());

이렇게 하면 안되는감?

Reply:
```suggestion
                intakeHistory.ifPresent(
                history -> history.modifyTargetAmount(intakeTargetAmountModifyRequest.toAmount())
        );
```
뭔가 불변 관련해서 헷갈리는데 
이런 식은 값 할당이 안 바뀌는 건지 잘 모르겟다

Reply:
이렇게 바꾸는건 어때요??

```java
@Transactional
public void modifyTarget(
        IntakeTargetAmountModifyRequest intakeTargetAmountModifyRequest,
        Long memberId
) {
    Member member = getMember(memberId);
    member.updateTargetAmount(intakeTargetAmountModifyRequest.toAmount());

    intakeHistoryRepository.findByMemberIdAndHistoryDate(memberId, LocalDate.now())
            .ifPresent(intakeHistory -> {
                intakeHistory.modifyTargetAmount(new Amount(intakeTargetAmountModifyRequest.amount()));
                intakeHistoryRepository.save(intakeHistory);
            });
}
```

1. `Amount`가 불변 객체라서 더티 체킹이 안 될 수도 있을 것 같아요 👉🏻 명시적으로 레파지토리 객체의 save 메서드 호출
2. 히로 코멘트처럼 request 객체에서 `amount` 필드를 바로 꺼내도 될 것 같아요

Reply:
```java
new Amount(intakeTargetAmountModifyRequest.amount())
```

```java
intakeTargetAmountModifyRequest.toAmount()
```

다시 생각해보니 이거 이렇게 써도 되지 않을까요?!

Reply:
# 정리
## 추론 결과
> 더티 체킹이 안 되었던 이유는 불변 객체 때문이 아니라, 엔티티 객체 자체가 **영속성 컨텍스트에 의해 관리되고 있지 않았기 때문**

### 더티체킹이 안 되는 경우
> 추측: 파라미터로 `Member` 객체가 넘어왔기 때문에, 영속성 컨텍스트에서 관리되는 객체가 아닌 단순 자바 객체로 다뤄짐

```java
@Transactional
public void modifyTarget(
        Member member,
        IntakeTargetAmountModifyRequest intakeTargetAmountModifyRequest
) {
    Amount updateAmount = intakeTargetAmountModifyRequest.toAmount();

    member.updateTargetAmount(updateAmount);
    // 💥 명시적으로 repository를 통해 save하지 않으면, 변경사항이 반영되지 않음 (= 더티 체킹 안 됨)
    // memberRepository.save(member); 

    // ...
}
```

### 더티체킹이 되는 경우
> `repository`에서 직접 꺼내온 엔티티는 영속성 컨텍스트에 등록되어 있으므로, 자동으로 더티 체킹 영역에 포함됨

```java
@Transactional
public void modifyTarget(
        Member member,
        IntakeTargetAmountModifyRequest intakeTargetAmountModifyRequest
) {
    // ...

    intakeHistoryRepository.findByMemberAndHistoryDate(member, LocalDate.now())
            .ifPresent(intakeHistory -> intakeHistory.modifyTargetAmount(updateAmount)); // 명시적으로 save 메서드를 호출하지 않더라도 더티체킹으로 인해 값이 정상적으로 수정됨
}
```